### PR TITLE
CTCP-4888: Added House Consignment Item level consignee

### PR DESCRIPTION
**Pull request by JakeGoodwin**
Screenshot 2024-02-29 at 16 59 31

### CODE REVIEW

File: app/pages/houseConsignment/index/items/ConsigneeAddressPage.scala [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+package pages.houseConsignment.index.items
+
+import models.{DynamicAddress, Index}
+import pages.QuestionPage
+import pages.sections.ItemsSection
+import play.api.libs.json.JsPath
+
+case class ConsigneeAddressPage(houseConsignmentIndex: Index, itemIndex: Index) extends QuestionPage[DynamicAddress] {
+
+  override def path: JsPath = ItemsSection(houseConsignmentIndex).path \ itemIndex.position \ "Consignee" \ toString
```
Comment (jackheslop96) [inline app/pages/houseConsignment/index/items/ConsigneeAddressPage.scala:26]: Can we define a consignee section using the ItemSection ?

File: app/utils/answersHelpers/ConsignmentAnswersHelper.scala [hunk 2]
```diff
@@ -360,11 +360,7 @@ class ConsignmentAnswersHelper(userAnswers: UserAnswers)(implicit messages: Mess
           val helper = new HouseConsignmentAnswersHelper(userAnswers, index)
           val rows = Seq(
             helper.consignorName,
-            helper.consignorIdentification,
-            helper.consigneeName,
-            helper.consigneeIdentification,
-            helper.consigneeCountry,
-            helper.consigneeAddress
+            helper.consignorIdentification
```
Comment (jackheslop96) [inline app/utils/answersHelpers/ConsignmentAnswersHelper.scala:363]: Where has the consignment consignee gone?

File: app/utils/answersHelpers/consignment/houseConsignment/ConsignmentItemAnswersHelper.scala [hunk 3]
```diff
@@ -20,8 +20,14 @@ import models.DocType.Previous
...
 import pages.NetWeightPage
-import pages.houseConsignment.index.items._
 import pages.houseConsignment.index.items.document.TypePage
+import pages.houseConsignment.index.items.{
+  ConsigneeAddressPage => ItemConsigneeAddressPage,
+  ConsigneeCountryPage => ItemConsigneeCountryPage,
+  ConsigneeIdentifierPage => ItemConsigneeIdentifierPage,
+  ConsigneeNamePage => ItemConsigneeNamePage,
+  _
+}
```
Comment (jackheslop96) [inline app/utils/answersHelpers/consignment/houseConsignment/ConsignmentItemAnswersHelper.scala:30]: Are these aliases necessary?

**Comment by jackheslop96** [comment]
> Screenshot 2024-02-29 at 16 59 31

@JakeGoodwin looking at the content doc it looks like the consignor rows should also have a Consignor h2. Do you know if this is being worked on by someone else?

File: app/utils/answersHelpers/ConsignmentAnswersHelper.scala (hunk 2 above)
Comment (JakeGoodwin) [reply app/utils/answersHelpers/ConsignmentAnswersHelper.scala:363]: It has been moved into it's own h2 as a section

**Comment by JakeGoodwin** [comment]
> > Screenshot 2024-02-29 at 16 59 31
> 
> @JakeGoodwin looking at the content doc it looks like the consignor rows should also have a Consignor h2. Do you know if this is being worked on by someone else?

I have spoken to Sayak about this, I think we're going to have a ticket raised to address this. This needs to be done in 3 places, the item level, HC level (XC page) and HC page

File: app/utils/answersHelpers/ConsignmentAnswersHelper.scala (hunk 2 above)
Comment (jackheslop96) [reply app/utils/answersHelpers/ConsignmentAnswersHelper.scala:363]: Ahh gotcha. I see now this is the house consignment consignee. All good

**Review by jackheslop96 — APPROVED** [review]

**Pull request merged**